experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/node.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TreeNode } from "../editableTreeTypes";
+import { SharedTreeNode } from "./types";
+
+export const nodeSym = Symbol("node");
+
+/**
+ * Exposes node functionality, such as event subscriptions, for the given SharedTreeNode.
+ * @alpha
+ */
+export function node(owner: SharedTreeNode): Required<SharedTreeNode>[typeof nodeSym] {
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	return owner[nodeSym]!;
```

Comment:
We've typically used the `?? fail("...")` pattern elsewhere in this package for these kinds of scenarios. I'm not sure how the performance compares, if that is important here.

Reply:
?? fail(..) sounds good.  Thanks.